component/main.jsonnet

```
@@ -1,10 +1,75 @@
 // main template for openshift4-terraform
 local kap = import 'lib/kapitan.libjsonnet';
-local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.openshift4_terraform;
 
-// Define outputs below
-{
-}
+local supported_providers = [
+  'exoscale',
+  'cloudscale',
+];
+
+local terraform_configs = {
+  cloudscale: {
+    'main.tf': {
+      module: {
+        cluster: params.terraform_variables,
+      },
+    },
+    'backend.tf': {
+      terraform: {
+        backend: {
+          http: {},
+        },
+      },
+    },
+    'output.tf': {
+      output: {
+        cluster_dns: {
+          value: '${module.cluster.dns_entries}',
+        },
+      },
+    },
+    'variables.tf': {
+      variable: {
+        ignition_bootstrap: {
+          default: '',
+        },
+      },
+    },
+  },
+  exoscale: {
+    'main.tf': {
+      module: {
+        cluster: params.terraform_variables,
+      },
+    },
+    'backend.tf': {
+      terraform: {
+        backend: {
+          http: {},
+        },
+      },
+    },
+    'output.tf': {
+      output: {
+        cluster_dns: {
+          value: '${module.cluster.ns_records}',
+        },
+      },
+    },
+    'variables.tf': {
+      variable: {
+        ignition_bootstrap: {
+          default: '',
+        },
+      },
+    },
+  },
+};
```

Might be worth deduplicating this by just having the diff as a local variable:

```suggestion
local cluster_dns = {
  cloudscale: '${module.cluster.dns_entries}',
  exoscale: '${module.cluster.ns_records}',
};

local terraform_config = {
  'main.tf': {
    module: {
      cluster: params.terraform_variables,
    },
  },
  'backend.tf': {
    terraform: {
      backend: {
        http: {},
      },
    },
  },
  'output.tf': {
    output: {
      cluster_dns: {
        value: cluster_dns[params.provider],
      },
    },
  },
  'variables.tf': {
    variable: {
      ignition_bootstrap: {
        default: '',
      },
    },
  },
};
```